src/io/sampling.jl

```diff
@@ -370,52 +370,116 @@ function fix_factor!(d::Domain; factors...)::Nothing
 end
 
 """
-    set_factor_bounds!(d::Domain, factor::Symbol, new_bnds::Tuple)::Nothing
-    set_factor_bounds!(d::Domain; factors...)::Nothing
+    get_bounds(dom::Domain, factor::Symbol)::Tuple
 
-Fix bounds of a parameter for sampling to those provided.
+Get factor bounds. If the factor is continuous, it will have a triangular distribution,
+then bounds will be a tuple with three elements (minimum, maximum, peak). If the factor is
+discrete, it will have a uniform distribution, then bounds will be a tuple with two elements
+(lower_bound, upper_bound). Note that, for discrete factors, the actual upper bound
+corresponds to the upper bound saved at the Domain's model_spec minus 1.0 because of how
+sampling works.
+
+# Arguments
+- `dom` : Domain
+- `factor` : Name of the factor to get the bounds from
+"""
+function get_bounds(dom::Domain, factor::Symbol)::Tuple
+    model::Model = dom.model
+    factor_filter::BitVector = collect(model[:fieldname]) .== factor
+
+    bounds = if _check_discrete(dom, factor)
+        lower, upper = model[:bounds][factor_filter][1]
+        (lower, upper - 1.0)
+    else
+        model[:bounds][factor_filter][1]
+    end
```

Comment:
I'm personally not a fan of these implicit return assignments.

Other than that, I suggest we stick to using [early returns](https://openfolder.sh/returning-early) to improve readability.

```julia
lower, upper = model[:bounds][factor_filter][1]
if _check_discrete(dom, factor)
    return (lower, upper - 1.0)
end

return (lower, upper)
```

Reply:
The first version of this code I wrote was precisely what you are suggesting. The problem is that sometimes bounds is a 3-values Tuple. But ok, let me try something else here then.

Reply:
Oh, use `getindex()`

```julia
bounds = model[...]
lower, upper = getindex.(bounds, (1, 2))
```

Might have to use `map` instead of `getindex()` but something like the above is what you want

Reply:
Huum, cool thanks for the tip!